numba_cuda/numba/cuda/tests/cudapy/test_extending.py

```diff
@@ -303,5 +303,35 @@ def foo(r, x):
         np.testing.assert_equal(r, x * 2)
 
 
+@skip_on_cudasim("Extensions not supported in the simulator")
+class TestArgHandlerRegistration(CUDATestCase):
```

Comment:
I think it would be wise to test with some other cases:

* Two arg handlers registered for different types - the test uses at least one argument of each type.
* One arg handler registered, one passed in `extensions=[...]` - does it behave as expected?
* Two arg handlers registered for the same type. What happens? An error? One displaces the other?
* Other things that could be thought of.